test/ui/browser/actions/test-close-tab.js

```diff
@@ -60,18 +76,37 @@ describe('Action - CLOSE_TAB', () => {
       'CLOSE_TAB on selected tab selects the next tab');
   });
 
-  it('Should destroy a selected tab and select previous when it is the last tab', function() {
+  it('Should destroy a selected tab and select previous when it is the last tab' +
+     ' and has no ancestor', function() {
     const { getCurrentPageIndex, getPages, dispatch } = this;
     const ids = getPages().map(p => p.id);
 
+    expect(getPages().get(3).ancestorId).toNotExist();
     dispatch(actions.closeTab(ids.get(3)));
+
     expect(getPages().size).toEqual(3);
     expect(getPages().get(2).location).toEqual('https://moz.org/2');
     expect(getCurrentPageIndex()).toEqual(2,
       'CLOSE_TAB on last selected tab selects the previous tab');
   });
 
-  it('Destroying the last tab should reset state', function() {
+  it('Should destroy a selected tab and select ancestor when it has an ancestor', function() {
```

Comment:
@bgrins -- test for ancestor selection is here.  I'll add a test to ensure that this doesn't fail when the ancestor has been closed.  We could also ask the UA service for *all* ancestors, and try to find the most recent one...